spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Resource.java

```diff
@@ -85,12 +82,9 @@ public S3Resource(Location location, S3Client s3Client, S3OutputStreamProvider s
 
 	@Override
 	public URL getURL() throws IOException {
-		List<String> splits = new ArrayList<>();
-		for (String split : location.getObject().split("/")) {
-			splits.add(URLEncoder.encode(split, StandardCharsets.UTF_8.toString()));
-		}
-		String encodedObjectName = String.join("/", splits);
-		return new URL("https", location.getBucket() + ".s3.amazonaws.com", "/" + encodedObjectName);
+		GetUrlRequest getUrlRequest = GetUrlRequest.builder().bucket(this.getLocation().getBucket())
```

Comment:
This change has caused an error in an integration test in Spring Cloud Config server's s3 support. The `location.getObject()` returns empty and is later complained about as an invalid key. `s3://test-bucket/` did not fail previously and does now.

Reply:
Hey @spencergibb ,
 
can you link me failing test so I can investigate and see what we can do about it?

Reply:
Yup, https://github.com/spring-cloud/spring-cloud-config/blob/b65f82180092c5f2e86f7476f6ef921d50c63229/spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/AwsS3IntegrationTests.java#L115-L124

Change this to 3.0.1 or 3.0.4 https://github.com/spring-cloud/spring-cloud-config/blob/b65f82180092c5f2e86f7476f6ef921d50c63229/pom.xml#L35

Reply:
Hey @spencergibb , I checked it out.
You are right, before it returned URL with host and file name just being /
when object name was null. Before:

```
return new URL("https", location.getBucket() + ".s3.amazonaws.com", "/" + encodedObjectName);
```

Since `encodedObjectName` will be empty string.

Now GetUrlRequest api call throws nullPtrException since object is null which makes sense from API standpoint.


I made [PR](https://github.com/awspring/spring-cloud-aws/pull/1158) 
this will solve issue for Spring Cloud Config. 

I am thinking is this correct way to handle it since AbstractResource says that `FileNotFoundExcpetion` should be thrown if url cannot be resolved and without object does it really makes sense to return url pointing only to the bucket or is it better to throw exception for users to handle it.

Something for me to think about. :)

Will try to get solution I am happy with and try to release soon!

Reply:
Thanks!  We will follow the new PR and keep an eye out for a release!